mongo/integration/mtest/mongotest.go

```diff
@@ -180,10 +180,16 @@ func New(wrapped *testing.T, opts ...*Options) *T {
 		t.createTestClient()
 	}
 
+	wrapped.Cleanup(t.Close)
+
 	return t
 }
 
-// Close cleans up any resources associated with a T. There should be one Close corresponding to every New.
+// Close cleans up any resources associated with a T.
+//
+// Deprecated: Close is now called automatically by [testing.T.Cleanup] at the
+// end of a test, so it does not need to be called explicitly. It is safe to
+// call multiple times.
 func (t *T) Close() {
```

Comment:
Do we still need to expose the `Close()`?

Reply:
No, there is now no use case for calling it. I waffled on un-exporting vs deprecating `Close` because we know there are external users of `mtest` who would probably be impacted by us un-exporting it. However, `mtest` is documented as unstable so we should be able to change it.

I will un-export it.